### DISCUSSION TITLE
The matrix-like alerts graph is fantastic

### DISCUSSION THREAD

**Discussion by maxEntropyProd** [jasonacox/Powerwall-Dashboard#209] [General]
Just a complement here. The "new" (not sure when it was implemented) graph that shows active alerts and their duration is brilliant! I've been having some issues with the PVS shutting down at times, and this will definitely help with fixing the problem as well as what may be causing the issues. Thanks!

**Comment by jasonacox** [comment]
Thanks @maxEntropyProd !!  Feedback via issues, requests, and advice is pure gold, but I confess, encouragement like your note is rocket fuel for the project and contributors! 🚀  Thanks for the note. 🙏 

For the Alert Graph, big credit is due to the amazing community and contributors who swarmed the idea in https://github.com/jasonacox/Powerwall-Dashboard/issues/158 including @DerickJohnson @BuongiornoTexas @youzer-name @mcbirse @BJReplay @wreiske.  Friends like these are what make OSS so fun and rewarding. 🤗 

If you still have it, can you paste a screenshot of the issue you saw?  It may help others trying to troubleshoot similar oddities.

**Reply by maxEntropyProd** [reply]
Hi @jasonacox The issue I've been having is that the PVS can go though periods where there are a lot of PVS power dropouts, but most of the time it's cloudy around here, so it's hard to tell if the dropout is due to some issue with the system, or just clouds. However, sometime the PVS shuts down altogether, so it's clear there are some issues. Today it was 100% sunny, yet there were numerous PVS dropouts in the morning (see 1st image). The new alerts graphic (2nd image) clearly shows that the dropouts correspond to a flurry of alters, many associated with communication. So, I decided to reboot our home network around 12:00, but I didn't touch anything regarding the Tesla solar/battery system. The interesting thing is that after to home network was rebooted, all the alters go away (a few linger after 12) and the PVS displays very stable power output thereafter. The Tesla gateway from the PW+ is connected to both the wired and wireless networks, and I'm wondering if that may be causing some collision problems. Anyway, having the alerts recorded with timestamps makes interpreting them much more straight forward.

![EnergyUse_20Mar2023](https://user-images.githubusercontent.com/16806730/226471560-4085624c-4f18-47ed-9452-d46850cfc6a4.jpg)
![Altertds_20Mar2023](https://user-images.githubusercontent.com/16806730/226471580-79bb7098-410c-4ca4-a9a9-99481a51656d.jpg)

**Reply by jasonacox** [reply]
HI @maxEntropyProd - that's quite the mess!  I have a theory. Rebooting your "home network" means that you were resetting your WiFi landscape.  How many of those "*MeterComms" devices are Neurio meters?  If they are, they are WiFi based.  I had a TERRIBLE time with my SolarMeterComms Neurio at first.  When it would go out, it would stop solar production.  Eventually Tesla replaced it and then one of the later Firmware updates stop shutting down solar production when it lost contact with the Neurio.  That was possible because I have a Powerwall+ system with the integrated Inverter and meter (separate from the Neurio).  However, your system may require those Comms updates to stay running.  The graphs above seem to align - Comms dropout and solar production dropout at the same time.

I suspect when you reset your home network, you cleared the WiFi channel collision that was happening. 

I think you "fixed" your problem but you will probably have to keep an eye on that.  At least with the Alerts graph, it is very clear when it is getting into a bad state.

* What firmware version are you running?
* Does your system use Neurio meters (and are they upgraded to firmware 1.7.2-Tesla)? - you can check using http://localhost:8675/vitals (where localhost is the address of the host running your dashboard).

**Reply by maxEntropyProd** [reply]
Hi @jasonacox. Yes, I think you're right. 

I'm running 22.36.4 for gateway, which was actually rolled back from 22.36.6 that occurred just a short time after (< 1 d) the 22.36.6 release. It is a PW+ with a Neurio, which is running 1.7.2-Tesla

In my router logs I often see several entries a week like "[WLAN access rejected: incorrect security] from MAC address ######", where the MAC address is the wifi of the PW+ inverter. However, when I check connected devices, the inverter wifi IS connected. I have no idea why the inverter wifi sometimes gives incorrect credentials, then other times it connects fine. Since I have a wired connection to the PW+ inverter, I have now disabled the inverter wifi connection. We'll see if this solves the problem. Thanks for you help!!

**Reply by jasonacox** [reply]
Nice! Smart move on disabling the WiFi.  Also, FYI, my system just updated to 22.36.7 tonight.

**Reply by maxEntropyProd** [reply]
Yes, same here 22.36.7, I also see the app got updated to 4.19.0. We'll see if disabling the inverter wifi solves the issues. 
I am curios about the Neurio though. I don't see it connected to my home network, and it's not clear what it's for. I have the Tesla Backup Switch, so it must monitor the current flow to/from the panel, and it's connected to the inverter via CAN bus. As far as I can tell, the Neuro in systems with the Gateway (not the backup switch) communicates wirelessly to the gateway, but it doesn't seem to be needed. The Neurio does show a solid blue/purple LED, which means it's connected to something though. I haven't looked at the Installer page on the gateway though.

**Reply by jasonacox** [reply]
In the Backup Gateway system that I have, the Neurio connects to the WiFi SSID that the gateway creates (the same one that you can connect to by scanning the QR code) that is not the home WiFi.  I suspect a similar thing happens for the Backup Switch.  But in any case, it would be competing with RF and other WiFi channels used by other SSIDs.
